<launch>
  <arg name="use_sim_time" default="true" />
  <arg name="gui" default="true" />
  <arg name="headless" default="false" />
  <arg name="world_name" default="$(find gazebo_radiation_plugins)/custom_models/reactor_room/updated_reactor_room2.world" />

  <!-- Short-term hack to support the original front_laser:=true argument for spawning
       the simulator. This will be removed in favour of using the config:=x arg instead. -->
  <arg name="front_laser" default="true" />
  <arg name="default_config" value="front_laser" if="$(arg front_laser)" />
  <arg name="default_config" value="base" unless="$(arg front_laser)" /> 
  <!-- end of hack -->

  <!--Load params -->
  <rosparam file="$(find gazebo_radiation_plugins)/custom_models/reactor_room/configs/attentuation.yaml" command="load"  />
  <rosparam file="$(find gazebo_radiation_plugins)/custom_models/reactor_room/configs/sensors.yaml" command="load"  />
  <rosparam file="$(find gazebo_radiation_plugins)/custom_models/reactor_room/configs/radiation.yaml" command="load"  />

  <node pkg="tf" type="static_transform_publisher" name="sim_sensor_to_robot" args="0 0 0 0 0 0 base_link sim_sensor 100"/>

  <!-- Launch Gazebo with the specified world -->

  <node name="gazebo" pkg="gazebo_radiation_plugins" type="gazebo"  output="screen"
  args="--verbose $(arg world_name)" />

  <!-- Configuration of Jackal which you would like to simulate.
       See jackal_description for details. -->
       <arg name="config" default="$(arg default_config)" />

  <!-- Load Jackal's description, controllers, and teleop nodes. -->
  <include file="$(find gazebo_radiation_plugins)/launch/custom_description.launch">
    <arg name="config" value="$(arg config)" />
  </include>
  <include file="$(find jackal_control)/launch/control.launch" />
  <include file="$(find jackal_control)/launch/teleop.launch">
    <arg name="joystick" value="false"/>
  </include>

  <!-- Spawn Jackal -->
  <node name="urdf_spawner" pkg="gazebo_ros" type="spawn_model" output="screen"
	      args="-urdf -model jackal -param robot_description -x -0 -y 0 -z 1.0" />

  <include file="$(find gazebo_radiation_plugins)/launch/gmapping_movebase.launch" />
  <include file="$(find gazebo_radiation_plugins)/launch/radmap.launch" />

  <node name="rviz" pkg="rviz" type="rviz"  args="-d $(find gazebo_radiation_plugins)launch/rviz_config/husky_radiation_mapping.rviz"/>

</launch>
